templates/docs.hamlet

```diff
@@ -0,0 +1,157 @@
+    By default, a hidden side-bar will be added to any 
+    <code>article
+    elements. Within these elements, each 
+    <code>p
+    and 
+    <code>pre
+    element will receive a small indicator, visible on hover. When clicked, the
+    side-bar will expand allowing visitors to comment on that specific portion
+    of your article.
+
+  <p>
+
+    Carnival also sends email notifications on replies and provides an RSS feed
+    of recent comments on your Site.
+
+  <p>
+
+    This page itself has Carnival loaded. Feel free to add comments and 
+    <a href=view-source:@{DocsR}>view-source
+    to see how it works.
+
+  <h2>Options
+
+  <p>
+    <code>Carnival.init
+    accepts an options object with any of the following keys:
+
+  <table .table>
+    <tr>
+      <th>Option
+      <th>Default Value
+      <th>Description
+    <tr>
+      <td><code>enabled</code>
+      <td><code>true</code>
+      <td>Add Carnival on this page?
+    <tr>
+      <td><code>article_author</code>
+      <td><code>""</code>
+      <td>The Article author. Used in the Comments Feed.
+    <tr>
+      <td><code>article_selector</code>
+      <td><code>"article"</code>
+      <td>The CSS selector used to find the Article element.
+    <tr>
+      <td><code>block_selector</code>
+      <td><code>":scope > p, :scope > pre"</code>
+      <td>
+        The CSS selector used to find elements within the Article on which to
+        add comment Threads. When this selector is used, 
+        <code>:scope
+        will refer to your Article element.
+    <tr>
+      <td><code>onNewComment</code>
+      <td><code>function(comment) { };</code>
+      <td>A callback invoked whenever comments are inserted in the DOM.
+
+  <p>
+    To pass options, you'll need to modify your event handler like so:
+
+  <pre>
+    <code>
+      \document.addEventListener("DOMContentLoaded", function() {
+      \  Carnival.init({
+      \    enabled: true,
+      \    article_author: '',
+      \    article_selector: 'article',
+      \    block_selector: ':scope > p, :scope > pre',
+      \    onNewComment: function(comment) { };
+      \  });
+      \});
+
+  <h2>Impact on the DOM
+
+  <p>
+
+    Carnival will insert an element to contain the comment threads and form.
+    This new element will be the first child of your Article element and have
+    the class 
+    <code>carnival
+    . To support the positioning of this element, Carnival must ensure that
+    your Article element is 
+    <code>position: relative
+    .
+
+  <p>
+
+    Carnival will add sibling elements after each commentable block to hold the
+    comment indicator and count. These elements will have the class 
+    <code>carnival-comment-indicator
+    .
+
+  <p>
+
+    When comments are opened, Carnival will add a 
+    <code>commenting
+    class to your Article element. To make room for itself, Carnival will shift
+    your Article element 
+    <code>150
+    pixels to its left. It does this by taking the element's current 
```

Comment:
Maybe "to the left"? Or is this text [irreplaceable](https://www.youtube.com/watch?v=2EwViQxSJJQ)?